dartagnan/src/main/java/com/dat3m/dartagnan/configuration/OptionInfo.java

```diff
@@ -0,0 +1,119 @@
+// Inspired from OptionCollector https://github.com/sosy-lab/java-common-lib
+//
+// SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+package com.dat3m.dartagnan.configuration;
+
+import com.google.common.reflect.ClassPath;
+import org.sosy_lab.common.configuration.Option;
+import org.sosy_lab.common.configuration.Options;
+
+import java.io.IOException;
+import java.lang.reflect.*;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+/**
+ * Collects all {@link Option}s of a program.
+ * <p>
+ * Mimics {@link org.sosy_lab.common.configuration.OptionCollector OptionCollector} with reduced functionality.
+ */
+public final class OptionInfo {
+
+    private static final Pattern IGNORED_CLASSES = Pattern.compile("^org\\.sosy_lab\\.common\\..*Test(\\$.*)?$");
```

Comment:
This is very nice, but somehow it "polutes" the output with the options coming from javaSMT. While it is true that we support those options and they (indirectly) affect our behavior, I think in general we won't use them and I would avoid showing then to the user. Can we use `IGNORED_CLASSES` to filter all options coming from them and not just test clases?

Reply:
We could also introduce a custom annotation on top of the usual `@Options`, and then we only look for classes that have this custom annotation attached. That is probably a lot easier than playing around with patterns. We could even have two help options then, one printing only our custom options and one that also prints all JavaSMT options.

Reply:
Alternatively, we filter our own classes and only show the options that we defined ourselves.

Reply:
We should definitely improve the description output provided by --help. One big problem is that the value ranges are not clear at all, especially for all our enums.

Reply:
I just pushed a change that shows enum options, but many descriptions can be improved. For all options it would be nice to show the default value (I did not find a nice way of doing this yet). I would also order the options alphabetically.

Reply:
As @xeren said, the default values must be extracted from the source code and can't be obtained via simple class information. So we would need to incorporate the default values directly into the description text (which is a little unfortunate).

Reply:
I think there are ways of [programmatically obtain the value of class fields](https://stackoverflow.com/questions/13400075/reflection-generic-get-field-value), however I think it would require to make the fields public which I'm not sure is a good design. 

I think manually incorporating this info is a no-go: it would be too easy to change the value and forget updating it in the description. Since the information is anyway logged, I would prefer not to give any such information in the help than risking giving a wrong one.

Reply:
You don't need to make the fields public. The problem is that you need an instance of a class to read the value of its field. But you won't be able to obtain such an instance. If the fields were static it would work without an instance.